src/Kernel/Deprecation.class.st

```diff
@@ -239,12 +239,13 @@ Deprecation >> transform [
 			replace: rule key with: rule value.
 		(rewriteRule executeTree: node)
 			ifFalse: [ ^ self signal ].
-		node replaceWith: rewriteRule tree. 
+		node replaceWith: (transformedNode := rewriteRule tree). 
 		Author 
 			useAuthor: 'AutoDeprecationRefactoring'
 			during: [aMethod origin compile: aMethod ast formattedCode classified: aMethod protocol].	
 		Log  ifNotNil: [:log | log add: self].
-		self logTranscript]
+		self logTranscript].
+	context return: (transformedNode evaluateForContext: context)
```

Comment:
Will not this re-evaluate the receiver and arguments? That would be wrong in case they have side effects... I think doing this properly is really complicated. We should diff the receiver/arguments nodes pre-rewriting with the rewritten ones, take the ones from the stack that match, and re-evaluate the ones that do not match. All this in the proper order and taking into account that arguments can be added and removed in addition of just modified.

Reply:
This complexity with no tests is a no go....

Reply:
And even! if the receiver/arguments do not match, nothing guarantees that a side effect in the rewritten nodes will overlap some other side effect of the previously executed ones.

Reply:
I think the best is to just restart the execution from some safe point

Reply:
Yes, true. This only works for the trival case (receiver and args are not expressions)